src/diagnostics.rs

```diff
@@ -321,7 +322,16 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 use NonHaltingDiagnostic::*;
                 let msg = match e {
                     CreatedPointerTag(tag) => format!("created tag {:?}", tag),
-                    PoppedPointerTag(item) => format!("popped tracked tag for item {:?}", item),
+                    PoppedPointerTag(item, access, tag) =>
+                        match tag {
+                            None => format!("popped tracked tag for item {:?}", item),
```

Comment:
Why is `access` ignored when `tag` is `None`?

Also, why not always use the 2nd arm here, in case the conflicting access uses an untagged ptr? Seems reasonable to say that explicitly?